### PR TITLE
Continue the v1.0 refactoring of Transformers and TransformersVision

### PR DESCRIPTION
**Pull request by RobinPicard**
I have not updated documentation yet, I will do it afterward and will tidy up the commits. A few questions or elements that need to be figured out:
- In your commit you had removed the methods `encode` and `decode` of the `TransformersTokenizer`. I put them back as it was not working without them. Why did you decide to remove them?
- With my latest commit, `TransformersVision` works but it does not let the user pass several images per prompt. It seems that it was possible before, so I still need to look into that.
- I removed the `device` parameter, considering that the user should use `device_map` as part of the kwargs. Do you think it's fine? I don't have a GPU so I have not tested it yet.

### CODE REVIEW

**Comment by rlouf** [comment]
I remove `encode` and `decode` to pass a logits processor instead.
